### PR TITLE
Custom dialogs; prepare for 2.0.0

### PR DESCRIPTION
**Pull request by stkent**
Addresses #128 and prepares for release of v2.0.0.

### CODE REVIEW

File: bugshaker/build.gradle [hunk 1]
```diff
@@ -4,42 +4,42 @@ buildscript {
...
 
     dependencies {
-        classpath 'com.github.ben-manes:gradle-versions-plugin:0.17.0'
-        classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.7.3'
-        classpath 'com.github.dcendents:android-maven-gradle-plugin:2.0'
-        classpath 'com.btkelly:gnag:2.3.0'
+        classpath "com.github.ben-manes:gradle-versions-plugin:0.17.0"
+        classpath "com.jfrog.bintray.gradle:gradle-bintray-plugin:1.7.3"
+        classpath "com.github.dcendents:android-maven-gradle-plugin:2.0"
+        classpath "com.btkelly:gnag:2.3.0"
```
Comment (btkelly) [inline bugshaker/build.gradle:10]: Gnag is out of date :) should be `2.4.1`

**Review by btkelly — APPROVED** [review]
Looks good, just the Gnag update which is mostly optional

File: bugshaker/src/main/java/com/github/stkent/bugshaker/flow/dialog/AppCompatDialogProvider.java [hunk 2]
```diff
@@ -31,11 +33,11 @@ public Dialog getAlertDialog(
...
-                .setTitle(ALERT_DIALOG_TITLE)
-                .setMessage(ALERT_DIALOG_MESSAGE)
-                .setPositiveButton(ALERT_DIALOG_POSITIVE_BUTTON, reportBugClickListener)
-                .setNegativeButton(ALERT_DIALOG_NEGATIVE_BUTTON, null)
-                .setCancelable(ALERT_DIALOG_CANCELABLE)
+                .setTitle(R.string.bugshaker_dialog_title)
+                .setMessage(R.string.bugshaker_dialog_message)
+                .setPositiveButton(R.string.bugshaker_dialog_positive_action, reportBugClickListener)
+                .setNegativeButton(R.string.bugshaker_dialog_negative_action, null)
+                .setCancelable(false)
```
Comment (rstockbridge) [inline bugshaker/src/main/java/com/github/stkent/bugshaker/flow/dialog/AppCompatDialogProvider.java:40]: much nicer! 👏

**Comment by rstockbridge** [comment]
What is the interaction between `AlertDialogType` and `DialogProvider`? Do they operate separately from each other?

**Comment by rstockbridge** [comment]
Aha I see "You may call either setAlertDialogType or setCustomDialogProvider, but not both."

**Comment by rstockbridge** [comment]
If you have the time, it might be useful to either include an option in the example application to illustrate a custom dialog, or make a second example application.

**Comment by rstockbridge** [comment]
This may not be a well-posed idea, but is there a way using types or something to make it impossible to call `setAlertDialogType` and `setCustomDialogProvider` both? Right now the you throw an exception after the fact, but I wonder if there's a way to enforce it upfront.

**Pull request merged**